server/api/impl/store/apps.go

```diff
@@ -30,14 +31,19 @@ func (s *Store) LoadApp(appID apps.AppID) (*apps.App, error) {
 	if v == nil {
 		return nil, utils.ErrNotFound
 	}
-	return apps.AppFromConfigMap(v), nil
+	app := apps.AppFromConfigMap(v)
+	app = s.populateAppWithManifest(app)
+	return app, nil
 }
 
 func (s *Store) StoreApp(app *apps.App) error {
 	conf := s.conf.GetConfig()
 	if len(conf.Apps) == 0 {
 		conf.Apps = map[string]interface{}{}
 	}
+	// do not store manifest in the config
+	app.ID = app.Manifest.AppID
+	app.Manifest = nil
```

Comment:
0/5 I'd advocate for complete separation of the `App` and `Manifest` structs. They represent different entities, and should not point at each other. The "client" should obtain them independently, as the use-case requires.

Reply:
why? An app has a manifest - it's kinda natural. Can App exist without a manifest? I think not.
Separating them is possible, but it will change a lot of code and we will be doing lot's of lookups as well.